app/controllers/relationships_controller.rb

```diff
@@ -0,0 +1,29 @@
+class RelationshipsController < ApplicationController
+  before_action :logged_in_user
+
+  def index
+    @user = User.find_by id: params[:id]
+    redirect_to root_path unless @user
+    @title = t ".#{params[:type]}"
+    @users = @user.send(params[:type]).page params[:page]
+    render "users/show_follow"
+  end
+
+  def create
+    @user = User.find_by id: params[:followed_id]
```

Comment:
add before_action for action 'index, create'
```
def find_user
    @user = User.find_by id: params[:id] || params[:followed_id]
    return if @user.present?
    
    flash[:danger] = t ".invalid_user"
    redirect_to root_path
  end
```

Reply:
commented